client/my-sites/activity/activity-log-tasklist/to-update.jsx

```diff
@@ -1,67 +1,59 @@
-import { get } from 'lodash';
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
-import { requestSiteAlerts } from 'calypso/state/data-getters';
+import { useRef } from 'react';
+import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import wpcom from 'calypso/lib/wp';
 import { getPluginsWithUpdates } from 'calypso/state/plugins/installed/selectors';
 import { isJetpackSiteSecondaryNetworkSite } from 'calypso/state/sites/selectors';
 
-const emptyList = [];
+const EMPTY_LIST = [];
 
 const unionBySlug = ( a = [], b = [] ) => [
 	...a,
 	...b.filter( ( be ) => ! a.some( ( ae ) => ae.slug === be.slug ) ),
 ];
 
-export default ( WrappedComponent ) => {
-	class ToUpdate extends Component {
-		static propTypes = {
-			siteId: PropTypes.number,
+export default function ( WrappedComponent ) {
+	function WithItemsToUpdate( { siteId, ...props } ) {
+		const pluginsWithUpdates = useSelector( ( state ) => {
+			if ( isJetpackSiteSecondaryNetworkSite( state, siteId ) ) {
+				return EMPTY_LIST;
+			}
+			return getPluginsWithUpdates( state, [ siteId ] ) ?? EMPTY_LIST;
+		} );
 
-			// Connected
-			plugins: PropTypes.arrayOf( PropTypes.object ),
-			themes: PropTypes.arrayOf( PropTypes.object ),
-			core: PropTypes.arrayOf( PropTypes.object ),
-		};
+		// merge the `pluginsWithUpdates` object into the previous value, so that plugins that
+		// have been updated are still part of the list and don't disappear.
+		const plugins = useRef( { siteId: undefined, mergedPlugins: undefined } );
+		const mergedPlugins =
+			plugins.current.siteId === siteId
+				? unionBySlug( pluginsWithUpdates, plugins.current.mergedPlugins )
+				: pluginsWithUpdates;
+		plugins.current = { siteId, mergedPlugins };
 
-		state = {
-			// Plugins already updated + those with pending updates
-			plugins: emptyList,
-			siteId: this.props.siteId,
-		};
+		const siteAlertsQuery = useQuery(
+			[ 'site-alerts', siteId ],
+			() => wpcom.req.get( { path: `/sites/${ siteId }/alerts`, apiNamespace: 'wpcom/v2' } ),
+			{ refetchInterval: 5 * 60 * 1000 }
```

Comment:
Should we disable `refetchOnWindowFocus` on this one?

Reply:
I added `{ staleTime: Infinity, refetchOnMount: 'always' }` options that have similar meaning: because the data never get stale, the automatic `refetchOnWindowFocus` or `refetchOnReconnect` or `refetchOnWhatever` conditions are never triggered, and `refetchOnMount` is set up to be always done, even if data are not stale.

That achieves the goal: refetch on mount and every 5 minutes. That keeps the tasklist stable, and items won't disappear from under your hands as you click "Update" and switch tabs back and forth.